Add Search Bar to Search for Cards

brijesh-0 opened the pull request:
**Description:**

This PR adds a search bar to filter cards, it looks in the card title, command, pretitle, and subtitle. As you type, the cards automatically update to show only the ones that match your search.

### Changes:

* Added a search bar for filtering.
* Set up JavaScript to filter the cards based on the input.

---

Most issues and PRs were for adding `git` commands so does this PR fit within the scope of the project?

<img width="1736" height="981" alt="1" src="https://github.com/user-attachments/assets/d0b986f3-f605-472c-a984-52ca1ed585b8" />
<img width="1736" height="981" alt="2" src="https://github.com/user-attachments/assets/df22ff68-ce01-4968-9687-c494a5554557" />

jtemporal commented on _layouts/home.html:
```diff
@@ -20,36 +20,58 @@
+  <!-- Search Bar -->
+  <div style="text-align: center; margin: 20px 0;">
+    <input type="text" id="search-input" placeholder="Search cards..." style="padding: 10px; width: 50%; max-width: 100%; border-radius: 8px;">
+  </div>
```
@brijesh-0 can we add the styling to the [css here](https://github.com/jtemporal/gitfichas/tree/main/assets/css) feel free to create a new `search.css` file to add that in

jtemporal commented:
Great PR thanks for putting it forward @brijesh-0 can you take a look at copilot’s suggestion and work on them I’ll be happily merge this once the adjustments are made. 😊

brijesh-0 commented:
Hey @jtemporal  , I've patched in the following changes:

- Implemented the no-results logic (using a counter) to display a message when no cards match the search.
- Extracted search bar styling to `search.css`.
- Left the minimal "no-results" styling inline for simplicity, but let me know if you prefer this be extracted too!

jtemporal commented:
Great @brijesh-0 i think we should extract the no result styling as well and add that to the search.css too so everything follows the same pattern for styling, what do you think?

brijesh-0 commented:
Done! Thanks for the input! @jtemporal
